src/iqstat.cpp

```diff
@@ -357,3 +234,137 @@ void usage() {
+
+    auto num_cols_selected = (brief ? output_format.brief_format_len
+                                    : output_format.long_format_len);
+
+    for (int i = 0; i < num_cols_selected; i++) {
+        addInxIval(&genQueryInp.selectInp, output_format.columns[i], 0);
+    }
+
+
+    if (!allFlag) {
+        char v1[BIG_STR];
+        snprintf( v1, BIG_STR, "='%s'", userName );
+        addInxVal(&genQueryInp.sqlCondInp, COL_RULE_EXEC_USER_NAME, v1);
+    }
+
+    std::string diagnostic;
+    if (ruleName != NULL && *ruleName != '\0') {
+        char v2[BIG_STR];
+        diagnostic = (boost::format( " (and matching key '%s')" ) % ruleName).str();
+        snprintf( v2, BIG_STR, "='%s'", ruleName );
+        addInxVal(&genQueryInp.sqlCondInp, COL_RULE_EXEC_ID, v2);
+    }
+
+    genQueryInp.maxRows = 50;
+
+    // idempotent version of clearGenQueryOut.
+    auto clear_gq_out = [&o = genQueryOut]{ clearGenQueryOut(o);
+                                            o->attriCnt = 0;
+                                          };
+
+    // TODO's : should we set attriCnt = 0 as the last thing in the clearGenQueryOut() routine?
+    // And...investigate 1. if this ^^^ is the best way or via init'ing via memset(ptr,0,sizeof(...)) 
+    //                   2. reimplementation of iqstat using the C++ query iterator
+    //                   3. possibility of memory leaks on rcGenQuery/rsGenQuery side
+
+    // RAII for both GenQuery variables.
+    auto cleanup_gq_in_out = [&, &i = genQueryInp, &o = genQueryOut]{ clearGenQueryInp(&i);
+                                                                      clear_gq_out();
+                                                                      free(o); o = nullptr; };
+    // Deploy the RAII.
+    const auto cleanup_gq = irods::at_scope_exit{[&] { cleanup_gq_in_out(); }};
+
+    int status = rcGenQuery( Conn, &genQueryInp, &genQueryOut );
+
+    if ( status == CAT_NO_ROWS_FOUND ) {
+        // Need to determine which table "no rows" refers to: USER or RULE.
+        namespace ia = irods::experimental::administration;
+        if (ia::client::exists(*Conn, ia::user{userName})) {
+            if ( allFlag ) {
+                printf( "No delayed rules pending%s\n", diagnostic.c_str() );
+            }
+            else {
+                printf( "No delayed rules pending for user %s%s\n", userName, diagnostic.c_str());
+            }
+            return 0;
+        }
+        else {
+            printf( "User %s does not exist.\n", userName );
+            return 0;
+        }
+    }
+    else if (status < 0) {
+        char* errno_message{};
+        const char* main_message = rodsErrorName(status, &errno_message);
+        std::cerr << "Error in rcGenQuery: " << main_message;
+        if (errno_message && *errno_message) {
+            std::cerr << " - "s + errno_message;
+        }
+        std::cerr << "\n";
+        std::free(errno_message);
+        return 0;
```

Comment:
Should this be something other than 0 to indicate the failure?

Reply:
Oh, you are right.  Do we have any set correspondence between return values and their meaning? Just curious.  I've seen lots of return 1; 2; or 3, but never any rationale.

Reply:
Returning 1 similar to handling of other iRODS errors.

Reply:
That will depend on the icommand, but in general, no.

If the meaning of the error code cannot be derived, use 1. It's more important that the user receive a non-zero error code than not at all.